prov/rxm/src/rxm_domain.c

```diff
@@ -214,6 +309,25 @@ rxm_cntr_open(struct fid_domain *fid_domain, struct fi_cntr_attr *attr,
 	return ret;
 }
 
+static int rxm_query_collective(struct fid_domain *domain,
+				enum fi_collective_op coll,
+				struct fi_collective_attr *attr,
+				uint64_t flags)
+{
+	struct rxm_domain *rxm_domain;
+
+	rxm_domain = container_of(domain, struct rxm_domain,
+				  util_domain.domain_fid);
+
+	if (!rxm_domain->util_coll_domain)
+		return -FI_ENOSYS;
+
+	return fi_query_collective(rxm_domain->util_coll_domain,
+				   coll, attr, flags);
+
+	// TODO: also check offload_coll_domain
```

Comment:
We're going to need to figure out how to handle this.  Do we report only the offload abilities or all of them?

Reply:
There are times we want all of them, like when we ran fabtests to exercise the util collectives. We could use a flag bit to indicate different expectation.

Reply:
All of them, as long as the offload provider is not configured/setup. Otherwise, only offload abilities.

Reply:
See #8271 to track requirements at the API level